File: src/breslin/clientside/application/states/applicationInitialize.cpp
//header
#include "applicationInitialize.h"

//log
#include "../../tdreamsock/dreamSockLog.h"

//state machine
#include "../../../statemachine/stateMachine.h"

//application
#include "../applicationBreslin.h"

/***************************************
*	CONSTRUCTORS		          
***************************************/
ApplicationInitialize::ApplicationInitialize(ApplicationBreslin* applicationBreslin)
{
	mApplication = applicationBreslin;
}

ApplicationInitialize::~ApplicationInitialize()
{

}

void ApplicationInitialize::enter()
{
	LogString("STATE: ApplicationInitialize");
}

void ApplicationInitialize::execute()
{
	//setup calls ogre specific graphics setup,
	//when it returns true we can begin our graphics stuff
	if (mApplication->setup())
	{
		mApplication->mSetup = true;
	}	

        if (mApplication->mSetup && mApplication->mConnected)
	{		
        	mApplication->mStateMachine->changeState(mApplication->mApplicationLogin);
	}
}

void ApplicationInitialize::exit()
{

}

File: src/breslin/clientside/application/states/application_initialize.php
var ApplicationInitialize = new Class(
{
	
initialize: function(application)
{
	this.mApplication = application;
},

enter: function()
{
},

execute: function()
{
        if (this.mApplication.setup())
        {
        	this.mApplication.mSetup = true;
        }
        if (this.mApplication.mSetup && this.mApplication.mConnected)
	{
		this.mApplication.hideLoadingScreen();
        	this.mApplication.mStateMachine.changeState(this.mApplication.mApplicationLogin);

	}
},

exit: function()
{
}

});
